### PR TITLE
Make yesod-auth buildable with GHC 9.0.1

### PR DESCRIPTION
**Pull request by geraldus**
Before submitting your PR, check that you've:

- [x] Bumped the version number
- [x] Documented new APIs with [Haddock markup](https://www.haskell.org/haddock/doc/html/index.html)
- [x] Added [`@since` declarations](http://haskell-haddock.readthedocs.io/en/latest/markup.html#since) to the Haddocks for new, public APIs

After submitting your PR:

- [x] Update the Changelog.md file with a link to your PR
- [x] Check that CI passes (or if it fails, for reasons unrelated to your change, like CI timeouts)

### CODE REVIEW

**Comment by geraldus** [comment]
All checks are passed, updating Changelog

File: yesod-auth/Yesod/Auth/Email.hs [hunk 1]
```diff
@@ -779,8 +782,8 @@ getPasswordR = do
     maid <- maybeAuthId
     case maid of
         Nothing -> loginErrorMessageI LoginR Msg.BadSetPass
-        Just _ -> do
-            needOld <- maybe (return True) needOldPassword maid
+        Just aid -> do
```
Comment (snoyberg) [inline yesod-auth/Yesod/Auth/Email.hs:785]: Nice

File: yesod-auth/Yesod/Auth/GoogleEmail2.hs [hunk 2]
```diff
@@ -239,24 +239,26 @@ authPlugin storeToken clientID clientSecret =
...
+                Left e  -> error e
                 Right t -> return t
 
         unless (tokenType' == "Bearer") $ error $ "Unknown token type: " ++ show tokenType'
 
         -- User's access token is saved for further access to API
         when storeToken $ setSession accessTokenKey accessToken'
 
-        personValue <- makeHttpRequest =<< personValueRequest token
+        personValReq <- personValueRequest token
```
Comment (snoyberg) [inline yesod-auth/Yesod/Auth/GoogleEmail2.hs:250]: I like the explicit `do`-notation usage here. Just curious: was this change needed for GHC 9?

**Review by snoyberg — APPROVED** [review]
Thanks!

File: yesod-auth/Yesod/Auth/GoogleEmail2.hs (hunk 2 above)
Comment (geraldus) [reply yesod-auth/Yesod/Auth/GoogleEmail2.hs:250]: Using bind led to polymorphism related error.  I was surprised by solution I eneded with:
```
• Couldn't match type: forall (m2 :: * -> *).
                           Yesod.Auth.MonadAuthHandler site0 m2 =>
                           m2 A.Value
                     with: m1 A.Value
      Expected: Request -> m1 A.Value
        Actual: Request -> AuthHandler site0 A.Value
    • In the first argument of ‘(=<<)’, namely ‘makeHttpRequest’
      In a stmt of a 'do' block:
        personValue <- makeHttpRequest =<< personValueRequest token
      In the expression:
        do mstate <- lookupGetParam "state"
           case mstate of
             Nothing -> invalidArgs [...]
             Just state -> do ...
           mcode <- lookupGetParam "code"
           code <- case mcode of
                     Nothing -> do ...
                     Just c -> return c
           ....
    |
250 |         personValue <-  makeHttpRequest =<< personValueRequest token
    |                         ^^^^^^^^^^^^^^^
```

Same for `hashAndSaltPassword = liftIO . saltPass` in `Email.hs`, eta reduce led to polymorphism error:
```
/Users/arthurfayzrakhmanov/Lab/haskell/build/yesod/yesod-auth/Yesod/Auth/Email.hs:244:27: error:
    • Couldn't match type: m0 Text
                     with: forall (m :: * -> *). MonadAuthHandler site m => m SaltedPass
      Expected: Text -> AuthHandler site SaltedPass
        Actual: Text -> m0 Text
    • In the expression: liftIO . saltPass
      In an equation for ‘hashAndSaltPassword’:
          hashAndSaltPassword = liftIO . saltPass
    • Relevant bindings include
        hashAndSaltPassword :: Text -> AuthHandler site SaltedPass
          (bound at Yesod/Auth/Email.hs:244:5)
    |
244 |     hashAndSaltPassword = liftIO . saltPass
    |                           ^^^^^^^^^^^^^^^^^
```

Comment (snoyberg) [reply yesod-auth/Yesod/Auth/GoogleEmail2.hs:250]: Cool, thanks for letting me know
